allennlp/models/reading_comprehension/document_qa.py

```diff
@@ -0,0 +1,401 @@
+
+            # TODO(joelgrus) figure out how to compute span_start_accuracy, span_end_accuracy,
+            # and span_accuracy in a context when there are multiple correct answers.
+            # (Conceptually it's not hard, but figure out how to do it using our metrics.)
+
+            output_dict["loss"] = loss
+
+        if metadata is not None:
+            output_dict['best_span_str'] = []
+            best_paragraphs = best_paragraph_word_span[:, 0]
+            best_span = best_paragraph_word_span[:, 1:]
+            batch_size = len(metadata)
+            for i in range(batch_size):
+                paragraph_idx = int(best_paragraphs[i].data.cpu().numpy())
+                passage_str = metadata[i]['paragraph_texts'][paragraph_idx]
+                offsets = metadata[i]['token_offsets'][paragraph_idx]
+                predicted_span = tuple(best_span[i].data.cpu().numpy())
+
+                start_offset = offsets[predicted_span[0]][0]
+                end_offset = offsets[predicted_span[1]][1]
+                best_span_string = passage_str[start_offset:end_offset]
+                output_dict['best_span_str'].append(best_span_string)
+                output_dict['best_span_str'].append(best_span_string)
+                answer_texts = metadata[i].get('answer_texts', [])
+                exact_match = f1_score = 0
+                if answer_texts:
+                    exact_match = squad_eval.metric_max_over_ground_truths(
+                            squad_eval.exact_match_score,
+                            best_span_string,
+                            answer_texts)
+                    f1_score = squad_eval.metric_max_over_ground_truths(
+                            squad_eval.f1_score,
+                            best_span_string,
+                            answer_texts)
+                self._official_em(100 * exact_match)
+                self._official_f1(100 * f1_score)
+        return output_dict
+
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+        return {'em': self._official_em.get_metric(reset),
+                'f1': self._official_f1.get_metric(reset)}
+
+    @staticmethod
+    def _get_best_span(span_start_logits: Variable, span_end_logits: Variable) -> Variable:
+        batch_size, num_paragraphs, passage_length = span_start_logits.size()
+        max_span_log_prob = [-1e20] * batch_size
+        # span_start_argmax = [0] * batch_size
+        best_word_span = Variable(span_start_logits.data.new()
+                                  .resize_(batch_size, 3).fill_(0)).long()
+
+        span_start_logits = span_start_logits.data.cpu().numpy()
+        span_end_logits = span_end_logits.data.cpu().numpy()
+
+        # pylint: disable=invalid-name
+        for b in range(batch_size):
+            for p in range(num_paragraphs):
+                span_start_argmax = 0
+                for j in range(passage_length):
```

Comment:
Is there anything stopping the span_start_argmax being more than j?
As far as I can see there is nothing stopping this?

Reply:
I don't understand the question.  The only assignment to `span_start_argmax` inside the loop is `span_start_argmax = j`.